rust/src/io/reader.rs

```diff
@@ -352,6 +354,16 @@ async fn _read_fixed_stride_array(
     .await
 }
 
+async fn read_null_array(
+    reader: &FileReader<'_>,
+    field: &Field,
+    batch_id: i32,
```

Comment:
This should accept `params` as well, please take a look of other `read_*_array` functions.

Mostly, this needs to support range query and `take(indices)`.

Reply:
Rectified. I had missed checking for other consumers of the `read_*_array` functions.